### PR TITLE
Chore: add linting dependency for ESLint, add PropTypes and nitpick fixes

### PR DESCRIPTION
**Pull request by Evalir**
Adds `PropTypes` to most components used, and tidies up the code a bit, also adding a missing ESLint dependency for linting to properly work on VSCode.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -53,6 +53,7 @@
     "eslint": "^6.8.0",
     "eslint-config-react-app": "^5.0.1",
     "eslint-config-react-tools": "^1.1.7",
+    "eslint-plugin-flowtype": "^4.7.0",
```
Comment (sohkai) [inline package.json]: Do we need this? We aren't using flow in this project at the moment.

File: components/ConversionForm/ConversionForm.js [hunk 2]
```diff
@@ -428,20 +432,29 @@ function Docs() {
...
           href="https://help.aragon.org/article/41-aragon-court"
           target="_blank"
+          rel="noopener noreferrer"
         >
           Docs
         </a>
       </li>
       <li>
-        <a href="https://court.aragon.org/dashboard" target="_blank">
+        <a
```
Comment (sohkai) [inline components/ConversionForm/ConversionForm.js]: We're doing this in a couple of places—time for a component :).

File: package.json (hunk 1 above)
Comment (Evalir) [reply package.json]: Yep, for some reason without this, the linter will scream on VSCode; I confirmed this with Andy earlier today.

**Comment by Evalir** [comment]
Let's do that!

**Comment by Evalir** [comment]
All done now! let's merge.

File: package.json [hunk 3]
```diff
@@ -53,12 +53,28 @@
...
+      "pre-commit": "lint-staged"
+    }
+  },
+  "lint-staged": {
+    "components/**/*.js": [
+      "eslint --fix"
+    ],
+    "lib/**/*.js": [
+      "eslint --fix"
+    ]
```
Comment (sohkai) [inline package.json]: Should we also add `pages`?

Comment (Evalir) [reply package.json]: While `pages` won't have much markup, sounds good to add it. 👍

**Pull request merged**